backend/kernelCI_app/unitTests/hardware_test.py

```diff
@@ -0,0 +1,215 @@
+from kernelCI_app.unitTests.utils.healthCheck import online
+from kernelCI_app.unitTests.utils.hardwareClient import HardwareClient
+from kernelCI_app.unitTests.utils.asserts import assert_status_code_and_error_response
+from kernelCI_app.utils import string_to_json
+from http import HTTPStatus
+from kernelCI_app.typeModels.hardwareDetails import HardwareDetailsPostBody
+
+
+client = HardwareClient()
+
+INVALID_HARDWARE = {
+    "id": "no hardware id",
+    "body": HardwareDetailsPostBody(
+        origin="",
+        startTimestampInSeconds=1737487800,
+        endTimestampInSeconds=1737574200,
+        selectedCommits={},
+        filter={},
+    ),
+}
+
+INVALID_REQUEST_BODY = {
+    "id": "google,juniper",
+    "body": HardwareDetailsPostBody(
+        origin="",
+        startTimestampInSeconds="",
+        endTimestampInSeconds="",
+        selectedCommits={},
+        filter={},
+    ),
+}
+
+GOOGLE_JUNIPER_HARDWARE_WITHOUT_FILTERS = {
+    "id": "google,juniper",
+    "body": HardwareDetailsPostBody(
+        startTimestampInSeconds=1740227400,
+        endTimestampInSeconds=1740659400,
+        selectedCommits={},
+        filter={},
+    ),
+}
+
+ARM_JUNO_HARDWARE_WITHOUT_FILTERS = {
+    "id": "arm,juno",
+    "body": HardwareDetailsPostBody(
+        startTimestampInSeconds=1740232800,
+        endTimestampInSeconds=1740664800,
+        selectedCommits={},
+        filter={},
+    ),
+}
+
+GOOGLE_JUNIPER_HARDWARE_WITH_FILTERS = {
+    "id": "google,juniper",
+    "body": HardwareDetailsPostBody(
+        startTimestampInSeconds=1740227400,
+        endTimestampInSeconds=1740659400,
+        selectedCommits={},
+        filter={"filter_architecture": ["arm64"], "filter_boot.status": ["PASS"]},
+    ),
+}
```

Comment:
In this case where you have two identical cases where some of the data should be always equal (like `id` and `startTimestampInSeconds` and `endTimestampInSeconds`, for example), but other parts can change you can just change that part from the first case.

```suggestion
GOOGLE_JUNIPER_HARDWARE_WITH_FILTERS = copy.deepcopy(GOOGLE_JUNIPER_HARDWARE_WITHOUT_FILTERS)
GOOGLE_JUNIPER_HARDWARE_WITH_FILTERS["body"].filter = {"filter_architecture": ["arm64"], "filter_boot.status": ["PASS"]}
```

in this case do not forget to import `copy`